src/Routing/PurlRouteProvider.php

```diff
@@ -163,7 +163,9 @@ public function getRouteCollectionForRequest(Request $request) {
         'query' => $query_parameters,
         'routes' => $routes,
       ];
-      $this->cache->set($cid, $cache_value, CacheBackendInterface::CACHE_PERMANENT, ['route_match']);
+      if ($routes->getResources()) {
```

Comment:
btw did you check why `getRoutesByPath` is returning incorrect route definitions in the first place?

Reply:
Its returning 2 routes "route:/test-subhojit/test-pro-drupal" and route:/test-subhojit/system/404
Not sure why.